### ISSUE TITLE
Upgrade the `http` crate to version 1.0.0

### ISSUE THREAD

**Issue by bruceg** [vectordotdev/vector#19179]
This requires updating several crates simultaneously, only some of which are currently in place. I will update this issue as the dependency crates are bumped:

- [X] `h2`
- [ ] `headers`
- [X] `http`
- [X] `http-body`
- [ ] `http-serde`
- [ ] `http-types`
- [X] `hyper`
- [ ] `hyper-proxy`
- [ ] `tonic`

**Comment by pront** [comment]
Seems like https://crates.io/crates/http-types and https://crates.io/crates/hyper-proxy are on the latest version.

Also, https://crates.io/crates/tokio-tungstenite ([PR](https://github.com/vectordotdev/vector/pull/19330)) is another crate that seems to affected by the http upgrade.

**Comment by bruceg** [comment]
Yes, both `http-types` and `hyper-proxy` do not have a released version that upgrades their usage to `http` 1.0. So, we are using their latest versions, but upgrading the other crates without those breaks the build.

Agreed, at least part of the `tokio-tungstenite` breakage is caused by this issue. I'll add that to the list.

**Comment by bruceg** [comment]
Note that `hyper-proxy` appears to be unmaintained, as the last commit was 3 years ago, which may require converting Vector to use `hyper-proxy2` ([link](https://crates.io/crates/hyper-proxy2)) or handling proxying manually.

**Comment by bruceg** [comment]
`tonic` now has released an updated version for the `http`/`hyper` 1.0 upgrade.
https://github.com/hyperium/tonic/blob/master/CHANGELOG.md

**Comment by pront** [comment]
Note: `http-proxy` is abandoned but https://crates.io/crates/hyper-http-proxy is straight up replacement for it.

**Comment by pront** [comment]
Interesting/related read: https://github.com/vectordotdev/vector/pull/22912#issuecomment-3064019563 and https://github.com/vectordotdev/vector/pull/22912#issuecomment-3255461959

**Comment by pront** [comment]
More motivation for upgrading:
* https://github.com/vectordotdev/vector/blob/master/deny.toml#L41-L52
* https://github.com/vectordotdev/vector/pull/25660

**Comment by pront** [comment]
Follow-up cleanup checklist for after all HTTP clients and servers have migrated to `http` 1 / Hyper 1. PR #26157 introduces some of the native implementation alongside temporary compatibility scaffolding, so completing component migrations is not the final step.

- [ ] Migrate remaining legacy paths within otherwise-migrated components. In particular, move ClickHouse Arrow schema discovery off `LegacyHttpClient` so ClickHouse constructs only one client.
- [ ] Promote `src/http/client_v1.rs` to the canonical, version-neutral HTTP client module and remove the Hyper 0.14 `HttpClient` / `HttpError` implementation.
- [ ] Promote `src/sinks/util/http_v1.rs` to the canonical sink HTTP service and remove the duplicate legacy `HttpService`, `HttpResponse`, request-builder, and response-retry implementations.
- [ ] Remove the retry compatibility glue added during migration: `HttpErrorClassify`, the generic error parameter on `HttpStatusRetryLogic`, and `new_with_error`.
- [ ] Make `http` 1 request/response types the native HTTP telemetry contract; remove `HttpRequestV1Telemetry`, `HttpResponseV1Telemetry`, and the header/version conversion helpers.
- [ ] Collapse the duplicated authentication helpers: remove the old `Auth` application path and rename `apply_v1` to the version-neutral API.
- [ ] Remove migration-only names and allowances (`client_v1`, `http_v1`, `LegacyHttpClient`, `*_v1`, and the associated `allow(dead_code)` attributes).
- [ ] Simplify the transport contract tests to exercise the single canonical client while retaining all direct, forward-proxy, CONNECT-proxy, authentication, TLS, streaming, and `NO_PROXY` scenarios.
- [ ] Rename the version-1 dependency aliases to their normal crate names and remove unused direct dependencies on `headers` 0.3, `http` 0.2, `http-body` 0.4, Hyper 0.14, `hyper-openssl` 0.9, and `hyper-proxy`. Older versions may remain transitively until their upstream users migrate.
- [ ] Run the complete unit, integration, and E2E suites after convergence, with explicit coverage for authenticated proxies and TLS through CONNECT.

`hyper_util::client::legacy::Client` is not part of this deletion list: despite its name, it is Hyper-util's high-level client API for Hyper 1, not a bridge to Hyper 0.14.
